### ISSUE TITLE
Field level validation must revalidate on component re-register.

### ISSUE THREAD

**Issue by theKashey** [erikras/redux-form#3500]
### Are you submitting a **bug report** or a **feature request**?
It is a bug
### What is the current behavior?
If you will change validation rule (or any other prop) Field will re-register itself.
### What is the expected behavior?
It will re-register itself and continue behave as it might behave.
### What's your environment?
7.0.4
### Other information

The bug is quite simple - 
I have form with 2 fields: The A and The B.
Validation Rule is simple: B must be equal A. And this is field level validation.

Then I'v change A it will call validation of B and display "SYNC_ERROR". Just after it I will provide a new validation function for B, and Field will de-register itself, wiping all the errors, and register back.

But this point it might call validation again, but it will not.
In other words - then I am providing a new field validation function to a Field - it must be called.

**Comment by HarrisGri** [comment]
I experience the same issue.

I have a field that is connected to a Country selection field. Which means, every time somebody changes a country the other field, that is connected to the Country selection will rerender and update his validators. Because depending on the selection the other field has to validate the postal_code depending on the country regulation.

**What currently happen's**
After the user selects an another payment country the field that is listening to that change is going to be rerendered and the _**validate**_ will be updated with the new _validate functions_ (redux-form will perform a @@redux-form/UNREGISTER_FIELD then @@redux-form/REGISTER_FIELD).

What happens after Unregister/Register is that all syncErrors are gone and the fields are marked as valid.

**Comment by danielrob** [comment]
Sandbox https://codesandbox.io/s/20635p081y

**Comment by atsu85** [comment]
Shouldn't this be closed, now that https://github.com/erikras/redux-form/pull/3909 is merged?

**Comment by danielrob** [comment]
@atsu85 as soon as 7.4.0 is released.
